### PR TITLE
feat(MultiSelect): Expose highlightedId and improve highlight handling for filterable

### LINKED ISSUE #1150: MultiSelect filterable - highlightedIndex not reset when highlighted option is filtered
Steps to reproduce:
- go to https://carbon-components-svelte.onrender.com/components/MultiSelect#filterable
- focus the input field
- press "arrow down" to highlight the "email" option (but do not select it)
- type "p" into the input field so all options are filtered away
- press Enter
- open browser console
- see `Uncaught TypeError: Cannot read properties of undefined (reading 'id') error` 

the error happens in this line https://github.com/carbon-design-system/carbon-components-svelte/blob/853942f998b656593afc8801970489aa10369ac9/src/MultiSelect/MultiSelect.svelte#L399

### PR DESCRIPTION
**Pull request by dominikg**
fixes #1150

### CODE REVIEW

File: src/MultiSelect/MultiSelect.svelte [hunk 1]
```diff
@@ -245,9 +251,10 @@
   $: checked = sortedItems.filter(({ checked }) => checked);
   $: unchecked = sortedItems.filter(({ checked }) => !checked);
   $: filteredItems = sortedItems.filter((item) => filterItem(item, value));
-  $: highlightedId = sortedItems[highlightedIndex]
-    ? sortedItems[highlightedIndex].id
-    : undefined;
+  $: highlightedId =
+    highlightedIndex > -1
+      ? (filterable ? filteredItems : sortedItems)[highlightedIndex]?.id ?? null
```
Comment (dominikg) [inline src/MultiSelect/MultiSelect.svelte:256]: is nullish coalescing ok here?

File: src/MultiSelect/MultiSelect.svelte [hunk 2]
```diff
@@ -394,23 +401,14 @@
...
-                    return { ...item, checked: !item.checked };
-                  });
-                } else {
-                  sortedItems = sortedItems.map((item, i) => {
-                    if (i !== highlightedIndex) return item;
-                    return { ...item, checked: !item.checked };
-                  });
-                }
+              if (highlightedId) {
+                const filteredItemIndex = sortedItems.findIndex(
```
Comment (dominikg) [inline src/MultiSelect/MultiSelect.svelte:405]: findIndex isn't supported by IE iirc. do you expect people to bring polyfills or should this be rewritten to be more compatible?

**Comment by dominikg** [comment]
happy to discuss this in a chat, find me on svelte discord :)

File: src/MultiSelect/MultiSelect.svelte (hunk 2 above)
Comment (metonym) [reply src/MultiSelect/MultiSelect.svelte:405]: We also use `find`. I think it's acceptable to focus on modern browsers. Also, MS will no longer support IE after June 15, 2022.

File: src/MultiSelect/MultiSelect.svelte [hunk 3]
```diff
@@ -156,6 +156,12 @@
    */
   export let selectionRef = null;
 
+  /**
+   * Id of the highlighted ListBoxMenuItem
+   * @type {null | string}
```
Comment (metonym) [inline src/MultiSelect/MultiSelect.svelte:161]: We should reuse the `MultiSelectItemId` typedef.

```suggestion
   * @type {null | MultiSelectItemId}
```

**Review by metonym — CHANGES_REQUESTED** [review]
@dominikg Thank you for fixing this!

Nice catch on the redundant `filterable` conditional.

My only requested change is to reuse the `MultiSelectItemId` type and re-run "yarn build:docs."

File: src/MultiSelect/MultiSelect.svelte (hunk 3 above)
Comment (dominikg) [reply src/MultiSelect/MultiSelect.svelte]: done

**Review by metonym — APPROVED** [review]
💯

**Comment by metonym** [comment]
Released in [v0.61.0](https://github.com/carbon-design-system/carbon-components-svelte/releases/tag/v0.61.0)

**Pull request merged**